doc/cbf/cbf_hld.md

```diff
@@ -0,0 +1,215 @@
+```
+
+Example:
+    127.0.0.1:6379[1]> hgetall  "CLASS_BASED_NEXT_HOP_GROUP:CbfNhg1"
+    1) "members"
+    2) "Nhg1,Nhg2,Nhg3,Nhg4"
+    3) "class_map"
+    4) "0:0,1:0,2:1,3:1,4:2,5:2,6:3,7:3"
+
+The ROUTE_TABLE is updated to allow the "nexthop_group" to allow both keys from NEXT_HOP_GROUP_TABLE and from the new CLASS_BASED_NEXT_HOP_GROUP_TABLE.
+```
+### ROUTE_TABLE
+    ;Stores a list of routes
+    ;Status: Mandatory
+    key           = ROUTE_TABLE:prefix
+    nexthop       = *prefix, ;IP addresses separated “,” (empty indicates no gateway)
+    ifname        = *PORT_TABLE.key,   ; zero or more separated by “,” (zero indicates no interface)
+    blackhole     = BIT ; Set to 1 if this route is a blackhole (or null0)
+    nexthop_group = NEXT_HOP_GROUP_TABLE.key or CLASS_BASED_NEXT_HOP_GROUP_TABLE.key ; index within the NEXT_HOP_GROUP_TABLE or CLASS_BASED_NEXT_HOP_GROUP_TABLE, optionally used instead of nexthop and intf fields
+ ```
+
+ The LABEL_ROUTE_TABLE is updated to allow the "nexthop_group" to allow both keys from NEXT_HOP_GROUP_TABLE and from the new CLASS_BASED_NEXT_HOP_GROUP_TABLE.
+```
+### LABEL_ROUTE_TABLE
+    ; Defines schema for MPLS label route table attributes
+    ;Status: Mandatory
+    key           = LABEL_ROUTE_TABLE:mpls_label ; MPLS label
+    nexthop       = STRING                   ; Comma-separated list of nexthops.
+    ifname        = STRING                   ; Comma-separated list of interfaces.
+    weight        = STRING                   ; Comma-separated list of weights.
+    nexthop_group = NEXT_HOP_GROUP_TABLE.key or CLASS_BASED_NEXT_HOP_GROUP_TABLE.key ; index within the NEXT_HOP_GROUP_TABLE or CLASS_BASED_NEXT_HOP_GROUP_TABLE, optionally used instead of nexthop and intf fields
+ ```
+
+### 3.2.2 CONFIG_DB
+In order to store the DSCP/EXP to FC mappings, 2 new CONFIG_DB tables will be added:
+
+```
+### DSCP_TO_FC_MAP
+    ;Stores a mapping between DSCP values and FC values. qos_map object with SAI_QOS_MAP_ATTR_TYPE == sai_qos_map_type_t::SAI_QOS_MAP_DSCP_TO_FC
+    ;Status: Mandatory
+    key        = DSCP_TO_FC_MAP_TABLE:string ; arbitrary string identifying the name of the map.
+    dscp_value = 1*DIGIT
+    fc_value   = 1*DIGIT
+```
+
+Example:
+    127.0.0.1:6379> hgetall "DSCP_TO_FC_MAP_TABLE:AZURE"
+     1) "3" ;dscp
+     2) "3" ;fc
+     3) "6"
+     4) "5"
+     5) "7"
+     6) "5"
+     7) "8"
+     8) "7"
+     9) "9"
+    10) "8"
+
+```
+### EXP_TO_FC_MAP
+    ;Stores a mapping between EXP values and FC values. qos_map object with SAI_QOS_MAP_ATTR_TYPE == sai_qos_map_type_t::SAI_QOS_MAP_EXP_TO_FC
+    ;Status: Mandatory
+    key        = EXP_TO_FC_MAP_TABLE:string ; arbitrary string identifying the name of the map.
+    exp_value  = 1*DIGIT
+    fc_value   = 1*DIGIT
+```
+
+Example:
+    127.0.0.1:6379> hgetall "EXP_TO_FC_MAP_TABLE:AZURE"
+     1) "3" ;exp
+     2) "3" ;fc
+     3) "6"
+     4) "5"
+     5) "7"
+     6) "5"
+     7) "8"
+     8) "7"
+     9) "9"
+    10) "8"
```

Comment:
Does orchagent checks for valid FC based on SAI_SWITCH_ATTR_MAX_NUMBER_OF_FORWARDING_CLASSES?

Reply:
Yes, it will.